### DISCUSSION TITLE
Add @template to model factories

### DISCUSSION THREAD

**Discussion by vpratfr** [laravel/framework#40071] [Ideas]
This would be a non-breaking change and would touch a single class: `Illuminate\Database\Eloquent\Factories\Factory`

There is currently no way for PHPStorm to know which precise model class would be returned by a call such as:

`$model = MyModel::factory()->create()`

As of now, PHPStorm would typehint the $model variable as either a `Collection` or a `Model`.

I propose to add the required `@template` phpdocs to the Factory class so that we can get proper typehinting in the IDEs and help static analysis tools to catch errors. In the above example, PHPStorm would know that `$model` is of the `MyModel` class if we define our factory with a phpdoc on top:

```
/**
 * @implements Factory<MyModel>
 */
class MyModelFactory extends Factory {
    // ...
}

$model = MyModel::factory()->create()
$model-> // IDE auto-completion proposes the MyModel members and functions
```

**Comment by crynobone** [comment]
Hey there,

We're closing this issue because it's inactive, already solved, old, or not relevant anymore. Feel free to open up a new issue if you're still experiencing this problem.
